ext/json/ext/generator/generator.c

```diff
@@ -913,6 +1092,22 @@ static void state_init(JSON_Generator_State *state)
 {
     state->max_nesting = 100;
     state->buffer_initial_length = FBUFFER_INITIAL_LENGTH_DEFAULT;
+    // TODO ADD RUNTIME CHECKS HERE?
+
+    switch(find_simd_implementation()) {
```

Comment:
I think you could do this in `Init_generator`, so it's only done once when the gem is required, rather than on every `JSON.generate` call.